cpp/src/join/mixed_join_common_utils.cuh

```diff
@@ -160,6 +161,42 @@ struct pair_expression_equality : public expression_equality<has_nulls> {
   }
 };
 
+/**
+ * @brief Equality comparator that composes two row_equality comparators.
+ */
+class double_row_equality {
+ public:
+  double_row_equality(row_equality equality_comparator, row_equality conditional_comparator)
+    : _equality_comparator{equality_comparator}, _conditional_comparator{conditional_comparator}
```

Comment:
Use `std::move` on the parameters to avoid copying.

Reply:
Indeed we can use `struct` with all public members so we won't even need a constructor.